### PR TITLE
docs: add contributing guidelines

### PR DESCRIPTION
**Pull request by ObserverOfTime**
May or may not need more stuff.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Contributing Guidelines
+
+## Create a grammar
+
+We recommended creating the grammar using our [template], either via the
+"Use this template" button on the web interface, or with the `--template` flag
```
Comment (amaanq) [inline CONTRIBUTING.md:6]:
```suggestion
"Use this template" button on the web interface, or via the `--template` flag
```
a bit clearer imo but totally doesn't matter

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+of the [gh cli]. Wait a moment for the initialization to complete before cloning
+the repository. The default license is MIT but you can change it to whatever you want.
+
+Make sure to read the Tree-sitter [documentation] before you start
+developing the grammar. The template handles most of the project setup
+for you. Just run these three commands in the grammar directory to get started:
```
Comment (amaanq) [inline CONTRIBUTING.md:12]:
```suggestion
for you. Just run these two commands in the grammar directory to get started:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+The template files include several `NOTE:` comments that you should read,
+act upon, and remove afterwards. You can view them all by running this command:
+
+```bash
+grep -Irn NOTE: --exclude-dir=node_modules
+```
+
+## Requirements for inclusion
```
Comment (amaanq) [inline CONTRIBUTING.md:33]:
```suggestion
## Requirements for Inclusion
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Contributing Guidelines
+
+## Create a grammar
```
Comment (amaanq) [inline CONTRIBUTING.md:3]:
```suggestion
## Creating a Grammar
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Requirements for inclusion
+
+If you'd like your grammar to be included in this organization you should adhere to the following:
+
+1. Include corpus tests to ensure the parser works as expected.
+2. The external scanner (if any) must be written in C and not C++.
+3. The grammar must be written in JavaScript and not TypeScript or anything else.
+4. The queries must use the last-wins precedence, a-la Neovim style.
+    - _(Optional)_ Include highlight queries and their tests. Other queries are not necessary.
```
Comment (clason) [inline CONTRIBUTING.md:41]: I would not make that optional.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    - The highlight captures should either use [Tree-sitter][STANDARD_CAPTURE_NAMES]'s
+      "standard" names, or [Neovim][treesitter-highlight-groups]'s names.
+5. You should be somewhat active in maintaining it; if there's no need for maintenance
+   that's totally okay, but you shouldn't let issues pile up for months to years on end.
+6. You must follow [SemVer] for your grammar: breaking changes must result in a major bump,
+   new features in a minor bump, and bug fixes in a patch bump.
+7. _(Optional)_ Commit messages should follow the [Conventional Commits] specification.
+8. Make sure all the metadata (names, links, versions, etc.) are correct and consistent.
+    - For example, the version in `package.json` must be the same as the one in `Cargo.toml`.
+9. _(Optional)_ Make use of our reusable [workflows] wherever possible.
```
Comment (clason) [inline CONTRIBUTING.md:51]: Same; we added them for a reason.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ObserverOfTime) [reply CONTRIBUTING.md:12]: I can't even remember what the third command was before I removed it…

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+3. The grammar must be written in JavaScript and not TypeScript or anything else.
+4. The queries must use the last-wins precedence, a-la Neovim style.
+    - _(Optional)_ Include highlight queries and their tests. Other queries are not necessary.
+    - The highlight captures should either use [Tree-sitter][STANDARD_CAPTURE_NAMES]'s
+      "standard" names, or [Neovim][treesitter-highlight-groups]'s names.
+5. You should be somewhat active in maintaining it; if there's no need for maintenance
+   that's totally okay, but you shouldn't let issues pile up for months to years on end.
+6. You must follow [SemVer] for your grammar: breaking changes must result in a major bump,
+   new features in a minor bump, and bug fixes in a patch bump.
+7. _(Optional)_ Commit messages should follow the [Conventional Commits] specification.
```
Comment (clason) [inline CONTRIBUTING.md:48]: Same. (Also, "should" already implies that it's not unconditionally required.)

Comment (clason) [reply CONTRIBUTING.md:48]: Lawyer-speak: there's a meaningful difference between "can", "should", and "must", and it helps to use these terms deliberately and consistently in any written guidance.

File: README.md [hunk 8]
```diff
@@ -6,16 +6,8 @@ find a grammar they're interested in.
 
 ## Contributing
 
-If you have a grammar you're interested in contributing, please open an issue [here](https://github.com/tree-sitter-grammars/.github/issues).
+If you want to create a grammar, or have already made one that you'd like to contribute,
+please read the [guidelines] and open an issue [here][issues] when your grammar is ready.
```
Comment (clason) [inline README.md:10]: Do we want to add a link to the Matrix room? Or create a new, non-Neovim branded, "parser club" room? (For non-Discordians)

File: CONTRIBUTING.md (hunk 7 above)
Comment (ObserverOfTime) [reply CONTRIBUTING.md:48]: Do we actually want to enforce this?

Comment (clason) [reply CONTRIBUTING.md:48]: Not sure. But it _does_ make semver easier (especially if you want to automate _that_).

Neovim has a commitlint workflow that does enforce this; it's not universally popular but does help. @dundargoc

**Review by amaanq — APPROVED** [review]

**Review by clason — APPROVED** [review]

**Pull request merged**